gomock/matchers.go

```diff
@@ -192,6 +210,12 @@ func Any() Matcher { return anyMatcher{} }
 //   Eq(5).Matches(4) // returns false
 func Eq(x interface{}) Matcher { return eqMatcher{x} }
 
+// Len returns a matcher that matches on length. This matcher returns false if
+// is compared to a type that is not an array, chan, map, slice, or string.
+func Len(i int) Matcher {
```

Comment:
nit: I'm imagining `mockFoo.EXPECT().Bar(gomock.Len(4))` which feels like it could be more natural but I can't think of a better alternative.